### ISSUE TITLE
Mac M1 chip error: Architecture mismatch found - arm64 expected . x86_64

### ISSUE THREAD

**Issue by hamza-choudhary** [lukaszkurantdev/react-native-fast-opencv#19]
**Error Description**
I am encountering an architecture mismatch error when trying to use the react-native-fast-opencv package on my Mac M1 system. The error I receive is as follows:

`Found architecture arm64; required architecture x86_64`

**System Information**
    Mac Model: MacBook Air M1
    macOS Version: 15.0.1
    Xcode Version: 16.0
    React Native Version: 0.75.4

**Steps Taken to Resolve the Issue**
    1) Added arm64 as an excluded architecture in Xcode settings.
    2) Tried excluding arm64 in the Podfile.
    3) Cleared caches, including derived data, node modules, and Pods.

I believe this issue could be related to compatibility between react-native-fast-opencv and Mac M1 architecture. Any insights or additional steps to resolve this issue would be greatly appreciated.

**Reproduction Steps**
    Set up a React Native project using react-native-fast-opencv.
    Attempt to build the project on a Mac M1 system for iOS.

If there is a known workaround or specific configuration needed to support M1 architecture, I would greatly appreciate any guidance.

**Error:**
<img width="857" alt="image(2)" src="https://github.com/user-attachments/assets/78f6df73-59a4-450e-8bcc-83dbcc590f1c">

**Detailed:**
<img width="1252" alt="Screenshot 2024-10-12 at 1 26 49 PM" src="https://github.com/user-attachments/assets/d600182e-5a97-42cf-9a8b-b2587110453a">

code:
```
import {useState} from 'react';
import {Button, Image, SafeAreaView} from 'react-native';
import {
  BorderTypes,
  DataTypes,
  ObjectType,
  OpenCV,
} from 'react-native-fast-opencv';
import {launchImageLibrary} from 'react-native-image-picker';
import {useRunOnJS, useWorklet} from 'react-native-worklets-core';

export function Home() {
  const [photo, setPhoto] = useState(null);
  const [result, setResult] = useState('');

  const getImageFromGallery = async () => {
    const result = await launchImageLibrary({
      mediaType: 'photo',
      includeBase64: true,
    });
    setPhoto(result.assets?.at(0) || null);
  };

  const setImage = useRunOnJS(data => {
    setResult(data);
  }, []);

  const worklet = useWorklet('default', () => {
    'worklet';
    if (photo?.base64) {
      const src = OpenCV.base64ToMat(photo.base64);
      const dst = OpenCV.createObject(ObjectType.Mat, 0, 0, DataTypes.CV_8U);
      const kernel = OpenCV.createObject(ObjectType.Size, 20, 20);
      const point = OpenCV.createObject(ObjectType.Point, 0, 0);

      OpenCV.invoke(
        'blur',
        src,
        dst,
        kernel,
        point,
        BorderTypes.BORDER_DEFAULT,
      );
      const dstResult = OpenCV.toJSValue(dst);

      setImage(dstResult.base64);

      OpenCV.clearBuffers(); // IMPORTANT
    }
  });

  return (
    <SafeAreaView style={{backgroundColor: 'white', flex: 1}}>
      <Button title="Select photo" onPress={getImageFromGallery} />
      <Button title="Process" onPress={() => worklet()} />

      {result && (
        <Image
          source={{uri: 'data:image/png;base64,' + result}}
          height={530}
          width={390}
        />
      )}
    </SafeAreaView>
  );
}

```

**Comment by lukaszkurantdev** [comment]
@hamza-choudhary  Hi, there is indeed a problem with building on a mac with ARM when you want to build a simulator version. For me these lines added to Podfile works:

```
post_install do |installer|
  # ...
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      config.build_settings['EXCLUDED_ARCHS[sdk=iphonesimulator*]'] = "arm64"
    end
  end
end
```

There that addition and `pod install` app builds correctly for the Simulator. 

But problem still exists, I need to investigate it more deeply.

**Comment by hamza-choudhary** [comment]
@lukaszkurantdev 
i have tried this but still its not working.
<img width="1258" alt="Screenshot 2024-10-13 at 11 58 16 AM" src="https://github.com/user-attachments/assets/275987e3-3cc7-445e-9587-3d27df3603b6">

**Comment by lukaszkurantdev** [comment]
It's related to Hermes, not this library. Make sure you've added `.xcode.env.local` or you have correct node path inside `.xcode.env`

**Comment by hamza-choudhary** [comment]
@lukaszkurantdev 
thanks for pointing out i just add hermes configuration in my Podfile now its working

for others who may face this error in future here is my current Podfile
```
require Pod::Executable.execute_command('node', ['-p',
  'require.resolve(
    "react-native/scripts/react_native_pods.rb",
    {paths: [process.argv[1]]},
  )', __dir__]).strip

min_ios_version_supported = '16.5'

platform :ios, min_ios_version_supported

prepare_react_native_project!

linkage = ENV['USE_FRAMEWORKS']
if linkage != nil
  Pod::UI.puts "Configuring Pod with #{linkage}ally linked Frameworks".green
  use_frameworks! :linkage => linkage.to_sym
end

$RNFirebaseAsStaticFramework = true

target 'paintTest' do
  config = use_native_modules!

  use_react_native!(
    :path => config[:reactNativePath],
    :hermes_enabled => true,
    :fabric_enabled => false,
    # An absolute path to your application root.
    :app_path => "#{Pod::Config.instance.installation_root}/.."
  )

  target 'paintTestTests' do
    inherit! :complete
  end

  post_install do |installer|
    react_native_post_install(installer)

    # Exclude arm64 architecture for simulators
    installer.pods_project.targets.each do |target|
      target.build_configurations.each do |config|
        config.build_settings['EXCLUDED_ARCHS[sdk=iphonesimulator*]'] = "arm64"
      end
    end
  end
end
```

**Comment by lukaszkurantdev** [comment]
After a lengthy investigation and numerous attempts, the bug and the need to add the above lines to the configuration in the Podfile has been fixed.

This was because the OpenCV dependency used on iOS did not include a compilation for the simulator on the ARM architecture. Unfortunately it appears that the official Pod is **no longer supported**.

Hence, I made the decision to create my own CocoaPod package, compiling OpenCV myself and creating a new Pod. With this:
- We have full control over the modules used in the library, and therefore the size of the compilation or the architectures needed.
- We also have control over the OpenCV version itself - in the case of the official Pod, the last available version was **4.3.0**, and in the current one it is **4.9.0.** I have not opted for the latest version **4.10.0**, as it contains numerous bugs related to processing and format conversion (JPEG -> PNG -> Matrix, etc.), but there is the possibility of creating another compilation in the future.
- The available architectures are now **ARM (iOS)**, **ARM (iOS Simulator)**, **X86_64 (iOS Simulator)**, hence there is no longer a need to add anything to the configuration.

The Pod package is available here: https://github.com/lukaszkurantdev/fastopencv-ios

The new Pod has been released in the new version `0.3.0`.

Closing this, @hamza-choudhary  thanks again for the issue.
